Xamarin.Forms.Core/Internals/NameScope.cs

```diff
@@ -23,6 +24,15 @@ void INameScope.RegisterName(string name, object scopedElement)
 			_names[name] = scopedElement;
 		}
 
+		//used by VS Live Visual Tree
+		internal string NameOf(object scopedObject)
+		{
+			var index = _names.Values.IndexOf(scopedObject);
```

Comment:
Linear search is not ideal. Hot Reload will call it for each element in visual tree.

Reply:
linear search on HR scenario is better than doing paying the perf of caching keys and values for every other scenarios (running apps, when HR is not plugged in)

Reply:
At a minimum you should do one linear search rather than two (first iterate thru values, then thru keys). A better way would be to enum _names, i.e.
`foreach(var entry in _names) { if (entry.Value == element) return entry.Key; } return null;`
I my little I see that double iteration does take twice the time.

I think realistically names elements will make up 10%-30% of all elements, i.e. adding another dictionary is not expensive memory usage wise. However perf gain will be significant. I my little app I creates 2,000 elements and named only 100 of them. Then looked up names for each of elements and aggregated total time. Note that because 1,900 elements did not have names we had to do full linear search for those elements. The numbers are:
  Total double linear search (current PR): 3.5ms
  Total single  linear search ("small fix" above): 1.5ms
  Total reverse dictionary lookup: 0.2ms
This is in a tiny stand alone app. In real app this might vary. I also tried different scenarios with high vs low percentage of names elements. On average linear search is 10 times slower.

Reply:
> I think realistically names elements will make up 10%-30% of all elements

that's a good point